spec/models/user_spec.rb

```diff
@@ -0,0 +1,21 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+
+  it 'is valid with a firstname and lastname' do
+    user = User.new(first_name: 'Shinya', last_name: 'Kitamura')
+    expect(user).to be_valid
+    expect(user.fullname).to eq 'Shinya Kitamura'
+  end
+
+  it 'is invalid without a firstname' do
+    user = User.new(first_name: nil, last_name: 'Kitamura')
+    expect(user).to be_invalid
+  end
+
+  it 'is invalid without a lastname' do
+    user = User.new(first_name: 'Shinya', last_name: nil)
+    expect(user).to be_invalid
+  end
```

Comment:
This could be organized better into contexts and let assignments:

```ruby
subject(:user) { build :user, first_name: first_name, last_name: last_name }

context "when both first name and last name are present" do
  let(:first_name) { "Shinya" }
  let(:last_name) { "Kitamura" }

  it { expect(user).to be_valid }
end

context "when only the last name is present" do
  let(:first_name) { nil }
  let(:last_name) { "Kitamura" }

  it { expect(user).to_not be_valid }
end

context "when only the first name is present" do
  let(:first_name) { "Shinya" }
  let(:last_name) { nil }

  it { expect(user).to_not be_valid }
end
```

What do you think?